plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -444,11 +444,27 @@ class GoogleBatchTaskHandler extends TaskHandler implements FusionAwareTask {
      * @return Retrieve the submitted task state
      */
     protected String getTaskState() {
-        final tasks = client.listTasks(jobId)
-        if( !tasks.iterator().hasNext() )
-            return 'PENDING'
-
         final now = System.currentTimeMillis()
+        final tasks = client.listTasks(jobId)
+        if( !tasks.iterator().hasNext() ) {
+            // if there are no tasks checks the job status
+            final jobStatus = client.getJobStatus(jobId);
+            final newState = jobStatus?.state as String
+            if (newState) {
+                taskState = newState
+                timestamp = now
+                if (newState == "FAILED"){
+                    final eventsCount = jobStatus.getStatusEventsCount()
+                    final lastEvent = eventsCount > 0 ? jobStatus.getStatusEvents(eventsCount - 1) : null
+                    if (lastEvent){
+                        log.warn1 "Batch job failure: ${lastEvent.getDescription()}"
+                    }
+                }
+                return taskState
+            } else {
+                return "PENDING"
+            }
```

Comment:
What about isolating this into a method `checkJobStatus()`, adding some tests whenever possible?

Reply:
I have isolated in a method and added a test. I have also modified jobError to manage the error in this situations. I see in this [part of the code](https://github.com/nextflow-io/nextflow/blob/fc6312856885fc818007a77d7e7e072e254bff32/plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy#L519) that if the error is due to an spot termination, it is throwing a ProcessException. 

Currently, I only print the error in the log and it is managed by as unknown error. 
```
Caused by:
  Process `FAILURE (1)` terminated for an unknown reason -- Likely it has been terminated by the external system

```
I could do the same as with spot looking for 'Batch error', but not sure if it will be correct. @pditommaso do you think leaving this errors as unexpected are correct, or should I generate a ProcessException as in the case of spot terminations?

Reply:
If the Google backend provides a more details error description could make sense to use the same approach as spot ie. [this](https://github.com/nextflow-io/nextflow/blob/51aea8f389adbaff8e820d226337c47587ad3c57/plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy#L519-L520)